src/ts/components/subtitletogglebutton.ts

```diff
@@ -57,7 +64,25 @@ export class SubtitleToggleButton extends ToggleButton<ToggleButtonConfig> {
         });
     }
 
+    private initPlayerEvents = () => {
+        this.player.on(this.player.exports.PlayerEvent.SubtitleAdded, this.checkSubtitleAvailability);
+        this.player.on(this.player.exports.PlayerEvent.SubtitleRemoved, this.checkSubtitleAvailability);
+        this.player.on(this.player.exports.PlayerEvent.PeriodSwitch, this.checkSubtitleAvailability);
+        this.player.on(this.player.exports.PlayerEvent.SourceUnloaded, this.checkSubtitleAvailability);
+    }
+
     private checkSubtitleAvailability = () => {
```

Comment:
I don't think the name is still fitting. Maybe something like `onAvailableSubtitlesChanged`